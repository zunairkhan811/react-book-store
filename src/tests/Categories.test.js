import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { render } from '@testing-library/react';
import MyCategory from '../components/Categories';

jest.mock('react-redux');
jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
}));
jest.mock('../redux/books/fetchApi', () => jest.fn());

describe('MyCategory Component', () => {
  beforeEach(() => {
    useDispatch.mockReturnValue(jest.fn());
    useSelector.mockReturnValue({
      data: [
        { category: 'Category 1' },
        { category: 'Category 2' },
        { category: 'Category 3' },
      ],
    });
  });

  it('should render MyCategory component', () => {
    const { container } = render(<MyCategory />);
    expect(container).toMatchSnapshot();
  });
});
