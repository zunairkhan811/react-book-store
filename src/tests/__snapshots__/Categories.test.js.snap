// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCategory Component should render MyCategory component 1`] = `
<div>
  <div
    class="category-container"
  >
    <ul
      class=""
    >
      <li
        role="presentation"
      >
        Category 1
      </li>
      <li
        role="presentation"
      >
        Category 2
      </li>
      <li
        role="presentation"
      >
        Category 3
      </li>
    </ul>
  </div>
</div>
`;
